Uno.Gallery/Uno.Gallery.Shared/Views/Shell.xaml.cs

```diff
@@ -213,34 +214,45 @@ private void NavigationViewControl_SizeChanged(object sender, SizeChangedEventAr
 
 		private void SamplesSearchBox_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
 		{
-			//This check can be removed when https://github.com/unoplatform/uno/issues/11635 is fixed
-#if !__ANDROID__ && !__IOS__
 			if(args.Reason != AutoSuggestionBoxTextChangeReason.UserInput)
 			{
 				return;
 			}
-#endif
 
 			if (string.IsNullOrEmpty(sender.Text))
 			{
 				sender.ItemsSource = null;
 				return;
 			}
 
-			var splitText = sender.Text.ToLower().ToLower().Split(" ");
+			var splitText = sender.Text.ToLower().Split(" ");
 
-			var samples = App.GetSamples()
+			var filteredSamples = App.GetSamples()
 				.OrderByDescending(x => x.SortOrder.HasValue)
 				.ThenBy(x => x.SortOrder)
 				.ThenBy(x => x.Title)
-				.Where(cat => splitText.All(key => cat.Title.IndexOf(key, StringComparison.OrdinalIgnoreCase) >= 0));
+				.Where(sample => splitText.All(key => sample.Title.Contains(key, StringComparison.OrdinalIgnoreCase)));
 
-			sender.ItemsSource = samples;
+			if (!filteredSamples.Any())
+			{
+				sender.ItemsSource = new List<Sample>() { new(new SamplePageAttribute(SampleCategory.None, "No suggestions found"), null) };
+			}
+			else
+			{
+				sender.ItemsSource = filteredSamples;
+			}
 		}
 
 		private void SamplesSearchBox_SuggestionChosen(AutoSuggestBox sender, AutoSuggestBoxSuggestionChosenEventArgs args)
 		{
-			(Application.Current as App)?.SearchShellNavigateTo(args.SelectedItem as Sample);
+			var sample = args.SelectedItem as Sample;
+
+			if (sample.Title.Contains("No suggestions found"))
```

Comment:
...

Can we at least use a constant?

Reply:
Good suggestion, @carldebilly, thanks. It's not the best, but it works better then before 😅